_posts/work/2018-01-30-hdf-in-the-cloud.md

```diff
@@ -64,40 +70,40 @@ So organizations like NASA are dumping large amounts of HDF onto Amazon's S3,
 that no one can actually read, except by downloading the entire file down to their hard drive,
 and then pulling out the particular bits that they need.
 This is inefficient.
-It misses out on the potential that cloud-hosted public data can offer our to society.
+It misses out on the potential that cloud-hosted public data can offer to our society.
 
 The rest of this post discusses a few of the options to solve this problem,
 including their advantages and disadvantages.
 
 1.  **Cloud Optimized GeoTIFF:** We can use modern and efficient formats from other domains, like Cloud Optimized GeoTIFF
 
-    **Good**: Fast, well established
+    *Good:* Fast, well established
 
-    **Bad**: Not actually sophisticated enough to handle some scientific use cases
+    *Bad:* Not actually sophisticated enough to handle some scientific use cases
 
 2.  **HDF + FUSE:** Continue using HDF, but mount cloud object stores as a file system with [Filesystem in Userspace, aka FUSE](https://en.wikipedia.org/wiki/Filesystem_in_Userspace)
 
-    **Good:** Works with existing files, no changes to the HDF library necessary
+    *Good:* Works with existing files, no changes to the HDF library necessary
 
-    **Bad:** It's complex, probably not as fast as possible, and has historically been brittle
+    *Bad:* It's complex, probably not as fast as possible, and has historically been brittle
 
 2.  **HDF + Custom Reader:** Continue using HDF, but teach it how to read from S3, GCS, ADL, ...
 
-    **Good:** Works with existing files, no complex FUSE tricks
+    *Good:* Works with existing files, no complex FUSE tricks
 
-    **Bad:** Requires changes to the HDF library and all downstream libraries (like Python wrappers).  Probably not performance optimal
+    *Bad:* Requires changes to the HDF library and all downstream libraries (like Python wrappers).  Probably not performance optimal
 
```

Comment:
This sounds like the S3VFD driver we are working on.  The VFD "Virtual File Driver" is a low-level plugin, so not a major change to the library.  Upstream libraries should not need much changes to utilize.  h5py already has the ability to select VFD on file open.

Reply:
> h5py already has the ability to select VFD on file open

It sounded like @llllllllll had to tweak h5py to make this work.  Is there another path for him?